### PR TITLE
Add gas limit to the runtime configuration

### PR DESCRIPTION
**Pull request by mehdi-aouadi**
## PR Description

Adds the gas limit to the runtime configuration (Required for the gas limit API)

## Fixed Issue(s)

https://github.com/ConsenSys/teku/issues/5999

## Documentation

- [ ] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [ ] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: validator/client/src/main/java/tech/pegasys/teku/validator/client/RuntimeProposerConfig.java [hunk 1]
```diff
@@ -68,20 +72,45 @@ public Optional<Eth1Address> getEth1AddressForPubKey(final BLSPublicKey pubKey)
...
+    return config.getFeeRecipient();
   }
 
-  public synchronized void addOrUpdate(
+  public Optional<UInt64> getGasLimitForPubKey(final BLSPublicKey publicKey) {
+    final Config config = proposerConfigMap.get(publicKey);
+    if (config == null) {
+      return Optional.empty();
+    }
+    return config.getGasLimit();
```
Comment (rolfyone) [inline validator/client/src/main/java/tech/pegasys/teku/validator/client/RuntimeProposerConfig.java]: may look neater if we 
```
final Optional<Config> maybeConfig = Optional.ofNullable(proposerConfigMap.get(publicKey));
return maybeConfig.map(Config::getGasLimit);
```

File: validator/client/src/test/java/tech/pegasys/teku/validator/client/RuntimeProposerConfigTest.java [hunk 2]
```diff
@@ -36,42 +37,69 @@ class RuntimeProposerConfigTest {
...
+    proposerConfig.addOrUpdateFeeRecipient(pubkey2, address2);
+    proposerConfig.addOrUpdateGasLimit(pubkey2, gasLimit2);
     final String data = Files.readString(testData);
     assertThat(data)
         .isEqualTo(
             String.format(
-                "{\"%s\":{\"fee_recipient\":\"%s\"},\"%s\":{\"fee_recipient\":\"%s\"}}",
-                pubkey, address, pubkey2, address2));
+                "{\"%s\":{\"fee_recipient\":\"%s\",\"gas_limit\":\"%s\"},\"%s\":{\"fee_recipient\":\"%s\",\"gas_limit\":\"%s\"}}",
+                pubkey, address, gasLimit, pubkey2, address2, gasLimit2));
```
Comment (rolfyone) [inline validator/client/src/test/java/tech/pegasys/teku/validator/client/RuntimeProposerConfigTest.java]: we could potentially have one test that saves both like here (just to 1 object)
then a second test that stores fee_recipient on one pubkey and gaslimit on a second, and then we have the 3 cases covered...

**Review by rolfyone — APPROVED** [review]
I only just saw this PR after reviewing the bigger one, but it's what I was hoping for.. It looks good :) a couple of nits, but a good size for reviewing easily, and a single concern, so happy to mark approved and you can consider the other couple of comments before merging 👍

File: validator/client/src/main/java/tech/pegasys/teku/validator/client/RuntimeProposerConfig.java (hunk 1 above)
Comment (mehdi-aouadi) [reply validator/client/src/main/java/tech/pegasys/teku/validator/client/RuntimeProposerConfig.java]: You mean `return maybeConfig.flatMap(Config::getGasLimit);` I guess :)

**Pull request merged**